libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1188,6 +1276,16 @@ export class RushConfiguration {
     return this._projects!;
   }
 
+  /**
+   * A list of all the available subspaces in this workspace.
+   */
+  public get subspaceNames(): string[] {
+    if (!this._projects) {
+      this._initializeAndValidateLocalProjects();
+    }
+    return Array.from(this._subspaceProjectsCache.keys());
+  }
```

Comment:
```suggestion
  public get subspaceNames(): Iterable<string> {
    if (!this._projects) {
      this._initializeAndValidateLocalProjects();
    }
    return this._subspaceProjectsCache.keys();
  }
```

Unless there's a strong reason to force the type into an array and copy the object upon using this method?